requirements_test_min.txt

```diff
@@ -1,3 +1,4 @@
 coverage~=7.0
 pytest
 pytest-cov~=4.0
+typing-extensions>=4.4.0
```

Comment:
Is this a test requirements or a runtime requirements ? Because we had an issue with this earlier, we want to know if astroid would fail with earlier version of typing_extensions see #1944. Maybe we need to test with the lower and higher version of ``typing_extensions `` and that would probably need to be handled with github actions and multiple jobs requiring both.

Reply:
It's a test requirement. `typing_extensions.TypeVar` was added in `4.4.0`.

> Maybe we need to test with the lower and higher version of `typing_extensions ` and that would probably need to be handled with github actions and multiple jobs requiring both.

Not sure it's worth it. This usually isn't an issue and the one time it is, we will see it fairly quickly.

There are alternatives
* We could bump the runtime requirement and make it unconditional.
* Or I could remove the test case I added here. It doesn't provide much value and I don't expect we'll introduce a regression anytime soon.

Reply:
> We could bump the runtime requirement and make it unconditional.

What do you mean by make it unconditional ? ``typing_extensions`` is used by a lot of lib, I think we need to use the less constrained requirement possible.

Reply:
> > We could bump the runtime requirement and make it unconditional.
> 
> What do you mean by make it unconditional ?

We don't install it for Python 3.11 currently. This here would need to be changed to
https://github.com/PyCQA/astroid/blob/c2519e52594fd5b971a0a1bf5094ec2fa307a7dd/pyproject.toml#L39

`"typing-extensions>=4.4.0"`

> `typing_extensions` is used by a lot of lib, I think we need to use the less constrained requirement possible.

I agree, however it shouldn't be an issue in this case. We would only require the latest version of `typing-extensions`.

Reply:
Ok, make sense, let's do this then.

Reply:
I tried something a bit different today with https://github.com/PyCQA/astroid/pull/1973/commits/b7c6b618a2d5305142ee661b442ccb785ee053a8. By moving the test case to a separate class and guarding it by `pytest.mark.skipif`, we can move the requirement to `requirements_test_brain.txt`. Those are only installed on linux and windows, not for PyPy. Maybe that is a bit better?

Let me know what you think. I can still move the requirement and update it in `pyproject.toml` like we talked about.

Reply:
Yeah it's actually nicer that way !